### PR TITLE
Add validation of schema for api-extensions

### PR DESCRIPTION
**Pull request by pope1838**
Depends on:
https://github.com/aml-org/amf-aml/pull/505

### CODE REVIEW

File: amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/common/parser/DeclaredAnnotationSchemaValidator.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+case class DeclaredAnnotationSchemaValidator(override val annotationIndex: Map[String, CustomDomainProperty])
+    extends AnnotationSchemaValidator {
+  override def validate(name: String, key: YNode, eh: AMFErrorHandler): Unit = {
+    annotationIndex.get(name).fold(addMissingAnnotationViolation(name, key, eh))(checkAnnotationSchema(_, eh))
+  }
+
+  private def checkAnnotationSchema(annotation: CustomDomainProperty, eh: AMFErrorHandler): Unit = {
+    annotation.schema match {
+      case any: AnyShape if any.meta != AnyShapeModel =>
```
Comment (tomsfernandez) [inline amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/common/parser/DeclaredAnnotationSchemaValidator.scala:23]: there is an auxliary method in AnyShape for this check

File: amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/common/parser/WebApiContext.scala [hunk 2]
```diff
@@ -220,15 +226,27 @@ abstract class WebApiContext(loc: String,
     if (isWarning) eh.warning(ClosedShapeSpecificationWarning, node, message, entry.location)
     else eh.violation(ClosedShapeSpecification, node, message, entry.location)
 
-  case class WebApiSemanticExtensionsFacadeBuilder() extends SemanticExtensionsFacadeBuilder {
+  case class WebApiSemanticExtensionsFacadeBuilder(annotationSchemaValidatorBuilder: AnnotationSchemaValidatorBuilder)
+      extends SemanticExtensionsFacadeBuilder {
     override def extensionName(name: String): SemanticExtensionsFacade = {
       val fqn = QName(name)
-      val dialect = if (fqn.isQualified) {
+      val facadeBuilder = if (fqn.isQualified) {
```
Comment (tomsfernandez) [inline amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/common/parser/WebApiContext.scala:233]: extract to method

File: amf-api-contract/shared/src/main/scala/amf/apicontract/internal/validation/definitions/ParserSideValidations.scala [hunk 3]
```diff
@@ -441,6 +458,12 @@ object ParserSideValidations extends Validations {
     ImplicitVersionParameterWithoutApiVersion.id -> all(WARNING), // TODO: should be violation
     InvalidVersionBaseUriParameterDefinition.id  -> all(WARNING), // TODO: should be violation
     HeaderMustBeObject.id                        -> Map(Async20Profile -> VIOLATION),
+    MissingAnnotationSchema.id -> Map(Raml10Profile -> VIOLATION,
```
Comment (tomsfernandez) [inline amf-api-contract/shared/src/main/scala/amf/apicontract/internal/validation/definitions/ParserSideValidations.scala:461]: GraphQL ?

**Review by tomsfernandez — DISMISSED** [review]

File: amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/common/parser/DeclaredAnnotationSchemaValidator.scala (hunk 1 above)
Comment (pope1838) [reply amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/common/parser/DeclaredAnnotationSchemaValidator.scala:23]: If you are talking about isAnyType of isNotExplicit isn't the same check.
The second one checks if there is a "type" key written, and is not my case.

File: amf-api-contract/shared/src/main/scala/amf/apicontract/internal/validation/definitions/ParserSideValidations.scala (hunk 3 above)
Comment (pope1838) [reply amf-api-contract/shared/src/main/scala/amf/apicontract/internal/validation/definitions/ParserSideValidations.scala:461]: Not for now, in fact graphql vendor is not in this branch (I added at another branch with graphql changes).

File: amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/common/parser/WebApiContext.scala (hunk 2 above)
Comment (pope1838) [reply amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/common/parser/WebApiContext.scala:233]: We already have this discussion...

Comment (tomsfernandez) [reply amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/common/parser/WebApiContext.scala:233]: sorry, though it was new

**Review by tomsfernandez — DISMISSED** [review]

**Review by arielmirra — APPROVED** [review]

**Pull request merged**